tests/testutils/container.go

```diff
@@ -77,6 +79,18 @@ func (container Container) WithCmd(cmd ...string) Container {
 	return container
 }
 
+func (container *Container) State(ctx context.Context) (*types.ContainerState, error) {
```

Comment:
Do you mind moving these down to the `*Container` receiver methods instead of being above the builder methods and update them to proxy to the underlying container instance after checking w/ `container.assertStarted(...)`?